patches/tModLoader/Terraria.ModLoader/ModWorld.cs

```diff
@@ -108,6 +108,88 @@ public virtual void ResetNearbyTileEffects()
+		}
+
+		/// <summary>
+		/// Gets called immedeately after all Players are updated, but before any NPCs get updated.
+		/// <para />
+		/// If you are looking too hook an earlier part of the update process, see <see cref="PreUpdateAnything" />.
+		/// If you are looking too hook a later part of the update process, see <see cref="MidUpdateNPCGore" />.
+		/// </summary>
+		public virtual void MidUpdatePlayerNPC()
+		{
+		}
+
+		/// <summary>
+		/// Gets called immedeately after all NPCs are updated, but before any Gores get updated.
+		/// <para />
+		/// If you are looking too hook an earlier part of the update process, see <see cref="MidUpdatePlayerNPC" />.
+		/// If you are looking too hook a later part of the update process, see <see cref="MidUpdateGoreProjectile" />.
+		/// </summary>
+		public virtual void MidUpdateNPCGore()
+		{
+		}
+
+		/// <summary>
+		/// Gets called immedeately after all Gores are updated, but before any Projectiles get updated.
+		/// <para />
+		/// If you are looking too hook an earlier part of the update process, see <see cref="MidUpdateNPCGore" />.
+		/// If you are looking too hook a later part of the update process, see <see cref="MidUpdateProjectileItem" />.
+		/// </summary>
+		public virtual void MidUpdateGoreProjectile()
+		{
+		}
+
+		/// <summary>
+		/// Gets called immedeately after all Projectiles are updated, but before any Items get updated.
+		/// <para />
+		/// If you are looking too hook an earlier part of the update process, see <see cref="MidUpdateGoreProjectile" />.
+		/// If you are looking too hook a later part of the update process, see <see cref="MidUpdateItemDust" />.
+		/// </summary>
+		public virtual void MidUpdateProjectileItem()
+		{
+		}
+
+		/// <summary>
+		/// Gets called immedeately after all Items are updated, but before any Dust gets updated.
+		/// <para />
+		/// If you are looking too hook an earlier part of the update process, see <see cref="MidUpdateProjectileItem" />.
+		/// If you are looking too hook a later part of the update process, see <see cref="MidUpdateDustTime" />.
+		/// </summary>
+		public virtual void MidUpdateItemDust()
+		{
+		}
+
+		/// <summary>
+		/// Gets called immedeately after all Dust is updated, but before Time effects get updated.
+		/// <para />
+		/// If you are looking too hook an earlier part of the update process, see <see cref="MidUpdateItemDust" />.
+		/// If you are looking too hook a later part of the update process, see <see cref="MidUpdateTimeWorld" />.
+		/// </summary>
+		public virtual void MidUpdateDustTime()
+		{
+		}
+
+		/// <summary>
+		/// Gets called immedeately after Time effects are updated, but before the World gets updated.
+		/// <para />
+		/// If you are looking too hook an earlier part of the update process, see <see cref="MidUpdateDustTime" />.
+		/// If you are looking too hook a later part of the update process, see <see cref="MidUpdateInvasionNet" />.
+		/// </summary>
+		public virtual void MidUpdateTimeWorld()
```

Comment:
Perhaps, "UpdateBetweenXAndY" moght be better?

Reply:
Yeah, I thought that the `MidUpdateXY` was nice and short whilst still conveying the meaning. And in cases where people aren't sure, they can always look at the documentation.
I was thinking that using plurals like `MidUpdatePlayersNPCs` might be better than `MidUpdatePlayerNPC`, but I'm not sure.

Reply:
I think that is up for debate/personal preference. Someone can also think "between Player and NPC means between any Player and NPC" which is basically the same thing as using plurals.